### PR TITLE
ADAPT-3713: unique IDs for listbox options, tabindex=0 for active option

### PR DESCRIPTION
**Pull request by andrewbaraniuk**
# NOT READY FOR REVIEW
- (Edit the above to reflect status)

# Summary
- TL;DR - what's this PR for?

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Navigate to...
3. Verify...

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: src/components/search/searchAutocomplete.js [hunk 1]
```diff
@@ -40,7 +49,7 @@ const searchAutocomplete = ({
             }}
             onFocus={(e) => setSelectedSuggestion(index)}
             aria-selected={selectedSuggestion === index ? "true" : "false"}
-            id="search-autocomplete-listbox"
+            id={searchAutocompleteListboxID()}
```
Comment (sherakama) [inline src/components/search/searchAutocomplete.js:52]: Why the addition of a random function? There are already two unique strings in this function within the `suggestion.objectID` and `index` variables. The random function seems like overhead to me unless someone knows something I don't.

Comment (yvonnetangsu) [reply src/components/search/searchAutocomplete.js:52]: Second what @sherakama said, if possible, please replace with the existing unique identifiers.

Comment (andrewbaraniuk) [reply src/components/search/searchAutocomplete.js]: Done!

Comment (yvonnetangsu) [reply src/components/search/searchAutocomplete.js]: Hmm I see that the suggestion.objectID sometimes has a white space if the suggest term has a white space in it
Screen Shot 2021-09-16 at 9 39 35 AM

Could we use the `index` variable instead perhaps?

**Comment by yvonnetangsu** [comment]
@andrewbaraniuk Please also add the id `search-autocomplete-listbox` to the `ul` element that contains the autocomplete list items, since that's referenced by the aria-controls attribute in the search input field

**Comment by yvonnetangsu** [comment]
Closing this as discussed in JIRA. I've merged @andrewbaraniuk 's work into my branch #172 and making some additional changes there. Thank you for the work, @andrewbaraniuk!

**Pull request closed without merging**